AvoidAsteroids/GameScene.swift

```diff
@@ -7,39 +7,292 @@
+    
     override func didMoveToView(view: SKView) {
         /* Setup your scene here */
-        let myLabel = SKLabelNode(fontNamed:"Chalkduster")
-        myLabel.text = "Hello, World!"
-        myLabel.fontSize = 45
-        myLabel.position = CGPoint(x:CGRectGetMidX(self.frame), y:CGRectGetMidY(self.frame))
+        let backgroundMusic = SKAudioNode(fileNamed: "music.m4a")
+        addChild(backgroundMusic)
+        
+        createPlayer()
+        
+        createSky()
+        
+        createBackground()
+        
+        createGround()
+        
+        createScore()
+        
+        createLogos()
+        
+        physicsWorld.gravity = CGVectorMake(0.0, -5.0)
+        physicsWorld.contactDelegate = self
         
-        self.addChild(myLabel)
     }
     
+    
     override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
        /* Called when a touch begins */
-        
-        for touch in touches {
-            let location = touch.locationInNode(self)
+        switch gameState {
+        case .ShowingLogo:
+            gameState = .Playing
             
-            let sprite = SKSpriteNode(imageNamed:"Spaceship")
+            let fadeOut = SKAction.fadeOutWithDuration(0.5)
+            let remove = SKAction.removeFromParent()
+            let wait = SKAction.waitForDuration(0.5)
+            let activatePlayer = SKAction.runBlock { [unowned self] in
+                self.player.physicsBody?.dynamic = true
+                self.initRocks()
+            }
             
-            sprite.xScale = 0.5
-            sprite.yScale = 0.5
-            sprite.position = location
+            let sequence = SKAction.sequence([fadeOut, wait, activatePlayer, remove])
+            logo.runAction(sequence)
             
-            let action = SKAction.rotateByAngle(CGFloat(M_PI), duration:1)
+        case .Playing:
+            player.physicsBody?.velocity = CGVectorMake(0, 0)
+            player.physicsBody?.applyImpulse(CGVectorMake(0, 20))
             
-            sprite.runAction(SKAction.repeatActionForever(action))
-            
-            self.addChild(sprite)
+        case .Dead:
+            let scene = GameScene(fileNamed: "GameScene")!
+            scene.scaleMode = .ResizeFill
+            let transition = SKTransition.moveInWithDirection(SKTransitionDirection.Right, duration: 1)
+            self.view?.presentScene(scene, transition: transition)
         }
     }
    
     override func update(currentTime: CFTimeInterval) {
         /* Called before each frame is rendered */
+        guard player != nil else { return }
+        let value = player.physicsBody!.velocity.dy * 0.001
+        let rotate = SKAction.rotateToAngle(value, duration: 0.1)
+        
+        player.runAction(rotate)
+    }
+    
+    func didBeginContact(contact: SKPhysicsContact) {
+        if contact.bodyA.node?.name == "scoreDetect" || contact.bodyB.node?.name == "scoreDetect" {
```

Comment:
Can you extract these string names into constants, I don't like seeing magic strings in the code